packages/rpc/src/endpoint.ts

```diff
@@ -215,10 +222,17 @@ export function createEndpoint<T>(
         break;
       }
       case FUNCTION_RESULT: {
-        const [callId] = data[1];
+        const [callId, method] = data[1];
 
-        callIdsToResolver.get(callId)!(...data[1]);
-        callIdsToResolver.delete(callId);
+        try {
+          callIdsToResolver.get(callId)!(...data[1]);
+          callIdsToResolver.delete(callId);
+        } catch (error) {
+          const {message} = error as Error;
+          throw new Error(
+            `Error in function result listener. Method: ${method} Error: ${message}`,
```

Comment:
I believe the second parameter in this case is a function ID, which will just be a random identifier. I don’t think this will be useful for you in debugging, although I guess there is no harm in providing it.